finagle-http/src/main/scala/com/twitter/finagle/http/DelayedReleaseService.scala

```diff
@@ -13,13 +27,38 @@ private[finagle] class DelayedReleaseService[-Req <: Request](
   service: Service[Req, Response]
 ) extends ServiceProxy[Req, Response](service) {
 
-  protected[this] val counter = new AsyncLatch
+  protected[this] val latch = new AsyncLatch
 
-  private[this] def proxy(in: Response) = {
+  override def apply(request: Req): Future[Response] = {
+    latch.incr()
+    service(request) transform {
+      case Return(r) if r.isChunked =>
+        Future.value(DelayedReleaseResponse(r, latch.decr))
```

Comment:
Yeah, I can revert this.  DelayedReleaseResponse seemed like a nice utility to have outside of this implementation, but now that I'm no longer using it in two places it seems best to merge back..